optimum/habana/transformers/generation/utils.py

```diff
@@ -573,6 +582,11 @@ def generate(
             or self._get_generation_mode(generation_config, assistant_model) == GenerationMode.BEAM_SEARCH
         )
         model_kwargs["bucket_size"] = generation_config.bucket_size if generation_config.static_shapes else -1
+        model_kwargs["reduce_recompile"] = (generation_config.reduce_recompile, False)[
+            generation_config.reduce_recompile is None
+        ]
```

Comment:
```suggestion
        model_kwargs["reduce_recompile"] = generation_config.reduce_recompile if generation_config.reduce_recompile is not None else False
```
I think it's easier to understand that way. You'll probably have to run `make style` too because this line is probably too long.